File: SearchSVC/clsSOURCETYPE.vb
Imports Microsoft.VisualBasic
Imports System.Configuration
Imports System
Imports System.Data.SqlClient
Imports System.Collections
Imports System.IO
Imports System.Data.Sql


Public Class clsSOURCETYPE


    '** DIM the selected table columns 
    Dim DB As New clsDatabaseSVR
    Dim DMA As New clsDmaSVR
    Dim UTIL As New clsUtilitySVR
    Dim LOG As New clsLogging



    Dim SourceTypeCode As String = ""
    Dim StoreExternal As String = ""
    Dim SourceTypeDesc As String = ""
    Dim Indexable As String = ""


    Dim SecureID As Integer = -1

    Sub New(ByVal iSecureID As Integer)
        SecureID = iSecureID
    End Sub


    '** Generate the SET methods 
    Public Sub setSourcetypecode(ByRef val as string)
        If Len(val) = 0 Then
            MsgBox("SET: Field 'Sourcetypecode' cannot be NULL.")
            Return
        End If
        val = UTIL.RemoveSingleQuotes(val)
        SourceTypeCode = val
    End Sub


    Public Sub setStoreexternal(ByRef val as string)
        If Len(val) = 0 Then
            val = "null"
        End If
        val = UTIL.RemoveSingleQuotes(val)
        StoreExternal = val
    End Sub


    Public Sub setSourcetypedesc(ByRef val as string)
        val = UTIL.RemoveSingleQuotes(val)
        SourceTypeDesc = val
    End Sub


    Public Sub setIndexable(ByRef val as string)
        If Len(val) = 0 Then
            val = "null"
        End If
        val = UTIL.RemoveSingleQuotes(val)
        Indexable = val
    End Sub






    '** Generate the GET methods 
    Public Function getSourcetypecode() As String
        If Len(SourceTypeCode) = 0 Then
            MsgBox("GET: Field 'Sourcetypecode' cannot be NULL.")
            Return ""
        End If
        Return UTIL.RemoveSingleQuotes(SourceTypeCode)
    End Function


    Public Function getStoreexternal() As String
        If Len(StoreExternal) = 0 Then
            StoreExternal = "null"
        End If
        Return StoreExternal
    End Function


    Public Function getSourcetypedesc() As String
        Return UTIL.RemoveSingleQuotes(SourceTypeDesc)
    End Function


    Public Function getIndexable() As String
        If Len(Indexable) = 0 Then
            Indexable = "null"
        End If
        Return Indexable
    End Function






    '** Generate the Required Fields Validation AS string 
    Public Function ValidateReqData() As Boolean
        If SourceTypeCode.Length = 0 Then Return false
        Return True
    End Function




    '** Generate the Validation AS string 
    Public Function ValidateData() As Boolean
        If SourceTypeCode.Length = 0 Then Return false
        Return True
    End Function




    '** Generate the INSERT AS string 
    Public Function Insert() As Boolean
        Dim b As Boolean = false
        Dim s As String = ""
        s = s + " INSERT INTO SourceType("
        s = s + "SourceTypeCode,"
        s = s + "StoreExternal,"
        s = s + "SourceTypeDesc,"
        s = s + "Indexable) values ("
        s = s + "'" + SourceTypeCode + "'" + ","
        s = s + StoreExternal + ","
        s = s + "'" + SourceTypeDesc + "'" + ","
        s = s + Indexable + ")"

        b = DB.DBExecuteSql(SecureID, S, False)
        If Not b Then
            'WriteTraceLog("clsSOURCETYPE : Insert : 01 : " + "ERROR: An unknown file type was NOT inserted. The SQL is: " + s)
            log.WriteToSqlLog("clsSOURCETYPE : Insert : 01 : " + "ERROR: An unknown file type was NOT inserted. The SQL is: " + s)
        End If
        Return b


    End Function




    '** Generate the UPDATE AS string 
    Public Function Update(ByVal WhereClause as string) As Boolean
        Dim b As Boolean = False
        Dim s As String = ""


        If Len(WhereClause) = 0 Then Return False


        s = s + " update SourceType set "
        's = s + "SourceTypeCode = '" + getSourcetypecode() + "'" + ", "
        s = s + "StoreExternal = " + getStoreexternal() + ", "
        s = s + "SourceTypeDesc = '" + getSourcetypedesc() + "'" + ", "
        s = s + "Indexable = " + getIndexable()
        WhereClause = " " + WhereClause
        s = s + WhereClause
        Return DB.DBExecuteSql(SecureID, S, False)
    End Function




    '** Generate the SELECT AS string 
    Public Function SelectRecs(ByVal SecureID As Integer) As SqlDataReader
        Dim b As Boolean = False
        Dim s As String = ""
        Dim rsData As SqlDataReader
        s = s + " SELECT "
        s = s + "SourceTypeCode,"
        s = s + "StoreExternal,"
        s = s + "SourceTypeDesc,"
        s = s + "Indexable "
        s = s + " FROM SourceType"

        Dim CS$ = DBgetConnStr() : Dim CONN As New SqlConnection(CS) : CONN.Open() : Dim command As New SqlCommand(s, CONN) : rsData = command.ExecuteReader()
        Return rsData
    End Function




    '** Generate the Select One Row AS string 
    Public Function SelectOne(ByVal SecureID As Integer, ByVal WhereClause as string) As SqlDataReader
        Dim b As Boolean = False
        Dim s As String = ""
        Dim rsData As SqlDataReader
        s = s + " SELECT "
        s = s + "SourceTypeCode,"
        s = s + "StoreExternal,"
        s = s + "SourceTypeDesc,"
        s = s + "Indexable "
        s = s + " FROM SourceType"
        s = s + WhereClause

        Dim CS$ = DBgetConnStr() : Dim CONN As New SqlConnection(CS) : CONN.Open() : Dim command As New SqlCommand(s, CONN) : rsData = command.ExecuteReader()
        Return rsData
    End Function




    '** Generate the DELETE AS string 
    Public Function Delete(ByVal WhereClause as string) As Boolean
        Dim b As Boolean = False
        Dim s As String = ""


        If Len(WhereClause) = 0 Then Return False


        WhereClause = " " + WhereClause


        s = " Delete from SourceType"
        s = s + WhereClause


        b = DB.DBExecuteSql(SecureID, S, False)
        Return b


    End Function




    '** Generate the Zeroize Table AS string 
    Public Function Zeroize() As Boolean
        Dim b As Boolean = False
        Dim s As String = ""


        s = s + " Delete from SourceType"


        b = DB.DBExecuteSql(SecureID, S, False)
        Return b


    End Function




    '** Generate Index Queries 
    Public Function cnt_PK34(ByVal SourceTypeCode As String) As Integer


        Dim B As Integer = 0
        Dim TBL$ = "SourceType"
        Dim WC$ = "Where SourceTypeCode = '" + SourceTypeCode + "'"


        B = DB.iGetRowCount(TBL$, WC)


        Return B
    End Function     '** cnt_PK34


    '** Generate Index ROW Queries 
    Public Function getRow_PK34(ByVal SourceTypeCode As String) As SqlDataReader


        Dim rsData As SqlDataReader = Nothing
        Dim TBL$ = "SourceType"
        Dim WC$ = "Where SourceTypeCode = '" + SourceTypeCode + "'"


        rsData = DB.GetRowByKey(SecureID, TBL$, WC)


        If rsData.HasRows Then
            Return rsData
        Else
            Return Nothing
        End If
    End Function     '** getRow_PK34


    ''' Build Index Where Caluses 
    '''
    Public Function wc_PK34(ByVal SourceTypeCode As String) As String


        Dim WC$ = "Where SourceTypeCode = '" + SourceTypeCode + "'"


        Return WC
    End Function     '** wc_PK34
End Class

File: Archiver/CS/clsSOURCETYPE.cs
// VBConversions Note: VB project level imports
using System.Collections.Generic;
using System;
using System.Drawing;
using System.Linq;
using System.Diagnostics;
using System.Data;
using Microsoft.VisualBasic;
using MODI;
using System.Xml.Linq;
using System.Collections;
using System.Windows.Forms;
// End of VB project level imports

using System.Configuration;
using System.Data.SqlClient;
using System.IO;
using System.Data.Sql;



namespace EcmArchiveClcSetup
{
	public class clsSOURCETYPE
	{
		
		
		//** DIM the selected table columns
		clsDatabase DB = new clsDatabase();
		clsDma DMA = new clsDma();
		clsUtility UTIL = new clsUtility();
		clsLogging LOG = new clsLogging();
		
		string SourceTypeCode = "";
		string StoreExternal = "";
		string SourceTypeDesc = "";
		string Indexable = "";
		
		
		
		
		//** Generate the SET methods
		public void setSourcetypecode(ref string val)
		{
			if (val.Length == 0)
			{
				MessageBox.Show("SET: Field \'Sourcetypecode\' cannot be NULL.");
				return;
			}
			val = UTIL.RemoveSingleQuotes(val);
			SourceTypeCode = val;
		}
		
		
		public void setStoreexternal(ref string val)
		{
			if (val.Length == 0)
			{
				val = "null";
			}
			val = UTIL.RemoveSingleQuotes(val);
			StoreExternal = val;
		}
		
		
		public void setSourcetypedesc(ref string val)
		{
			val = UTIL.RemoveSingleQuotes(val);
			SourceTypeDesc = val;
		}
		
		
		public void setIndexable(ref string val)
		{
			if (val.Length == 0)
			{
				val = "null";
			}
			val = UTIL.RemoveSingleQuotes(val);
			Indexable = val;
		}
		
		
		
		
		
		
		//** Generate the GET methods
		public string getSourcetypecode()
		{
			if (SourceTypeCode.Length == 0)
			{
				MessageBox.Show("GET: Field \'Sourcetypecode\' cannot be NULL.");
				return "";
			}
			return UTIL.RemoveSingleQuotes(SourceTypeCode);
		}
		
		
		public string getStoreexternal()
		{
			if (StoreExternal.Length == 0)
			{
				StoreExternal = "null";
			}
			return StoreExternal;
		}
		
		
		public string getSourcetypedesc()
		{
			return UTIL.RemoveSingleQuotes(SourceTypeDesc);
		}
		
		
		public string getIndexable()
		{
			if (Indexable.Length == 0)
			{
				Indexable = "null";
			}
			return Indexable;
		}
		
		
		
		
		
		
		//** Generate the Required Fields Validation method
		public bool ValidateReqData()
		{
			if (SourceTypeCode.Length == 0)
			{
				return false;
			}
			return true;
		}
		
		
		
		
		//** Generate the Validation method
		public bool ValidateData()
		{
			if (SourceTypeCode.Length == 0)
			{
				return false;
			}
			return true;
		}
		
		//** Generate the INSERT method
		public bool Insert()
		{
			bool b = false;
			string s = "";
			s = s + " INSERT INTO SourceType(";
			s = s + "SourceTypeCode,";
			s = s + "StoreExternal,";
			s = s + "SourceTypeDesc,";
			s = s + "Indexable) values (";
			s = s + "\'" + SourceTypeCode + "\'" + ",";
			s = s + StoreExternal + ",";
			s = s + "\'" + SourceTypeDesc + "\'" + ",";
			s = s + Indexable + ")";
			
			b = DB.ExecuteSqlNewConn(s, false);
			if (! b)
			{
				LOG.WriteToArchiveLog((string) ("clsSOURCETYPE : Insert : 01 : " + "ERROR: An unknown file type was NOT inserted. The SQL is: " + s));
				LOG.WriteToArchiveLog((string) ("clsSOURCETYPE : Insert : 01 : " + "ERROR: An unknown file type was NOT inserted. The SQL is: " + s));
			}
			return b;
			
			
		}
		
		
		//** Generate the UPDATE method
		public bool Update(string WhereClause)
		{
			bool b = false;
			string s = "";
			
			
			if (WhereClause.Length == 0)
			{
				return false;
			}
			
			
			s = s + " update SourceType set ";
			//s = s + "SourceTypeCode = '" + getSourcetypecode() + "'" + ", "
			s = s + "StoreExternal = " + getStoreexternal() + ", ";
			s = s + "SourceTypeDesc = \'" + getSourcetypedesc() + "\'" + ", ";
			s = s + "Indexable = " + getIndexable();
			WhereClause = (string) (" " + WhereClause);
			s = s + WhereClause;
			return DB.ExecuteSqlNewConn(s, false);
		}
		
		
		
		
		//** Generate the SELECT method
		public SqlDataReader SelectRecs()
		{
			bool b = false;
			string s = "";
			SqlDataReader rsData;
			s = s + " SELECT ";
			s = s + "SourceTypeCode,";
			s = s + "StoreExternal,";
			s = s + "SourceTypeDesc,";
			s = s + "Indexable ";
			s = s + " FROM SourceType";
			
			string CS = DB.getGateWayConnStr(modGlobals.gGateWayID);
			SqlConnection CONN = new SqlConnection(CS);
			CONN.Open();
			SqlCommand command = new SqlCommand(s, CONN);
			rsData = command.ExecuteReader();
			return rsData;
		}
		
		
		
		
		//** Generate the Select One Row method
		public SqlDataReader SelectOne(string WhereClause)
		{
			bool b = false;
			string s = "";
			SqlDataReader rsData;
			s = s + " SELECT ";
			s = s + "SourceTypeCode,";
			s = s + "StoreExternal,";
			s = s + "SourceTypeDesc,";
			s = s + "Indexable ";
			s = s + " FROM SourceType";
			s = s + WhereClause;
			
			string CS = DB.getGateWayConnStr(modGlobals.gGateWayID);
			SqlConnection CONN = new SqlConnection(CS);
			CONN.Open();
			SqlCommand command = new SqlCommand(s, CONN);
			rsData = command.ExecuteReader();
			return rsData;
		}
		
		
		
		
		//** Generate the DELETE method
		public bool Delete(string WhereClause)
		{
			bool b = false;
			string s = "";
			
			
			if (WhereClause.Length == 0)
			{
				return false;
			}
			
			
			WhereClause = (string) (" " + WhereClause);
			
			
			s = " Delete from SourceType";
			s = s + WhereClause;
			
			
			b = DB.ExecuteSqlNewConn(s, false);
			return b;
			
			
		}
		
		
		
		
		//** Generate the Zeroize Table method
		public bool Zeroize()
		{
			bool b = false;
			string s = "";
			
			
			s = s + " Delete from SourceType";
			
			
			b = DB.ExecuteSqlNewConn(s, false);
			return b;
			
			
		}
		
		
		
		
		//** Generate Index Queries
		public int cnt_PK34(string SourceTypeCode)
		{
			
			
			int B = 0;
			string TBL = "SourceType";
			string WC = "Where SourceTypeCode = \'" + SourceTypeCode + "\'";
			
			
			B = DB.iGetRowCount(TBL, WC);
			
			
			return B;
		} //** cnt_PK34
		
		
		//** Generate Index ROW Queries
		public SqlDataReader getRow_PK34(string SourceTypeCode)
		{
			
			
			SqlDataReader rsData = null;
			string TBL = "SourceType";
			string WC = "Where SourceTypeCode = \'" + SourceTypeCode + "\'";
			
			
			rsData = DB.GetRowByKey(TBL, WC);
			
			
			if (rsData.HasRows)
			{
				return rsData;
			}
			else
			{
				return null;
			}
		} //** getRow_PK34
		
		
		/// Build Index Where Caluses
		///
		public string wc_PK34(string SourceTypeCode)
		{
			
			
			string WC = "Where SourceTypeCode = \'" + SourceTypeCode + "\'";
			
			
			return WC;
		} //** wc_PK34
	}
	
}
